src/RefractDataStructure.cc

```diff
@@ -514,51 +663,69 @@ namespace drafter {
         }
 
-        TransformElementData(element, data);
+        TransformElementData(element, data, value.hasSourceMap());
 
         return element;
     }
 
     template <typename T>
-    refract::MemberElement* RefractElementFromProperty(const mson::PropertyMember& property, const mson::BaseTypeName defaultNestedType)
+    refract::MemberElement* RefractElementFromProperty(const SectionInfo<mson::PropertyMember>& property, const mson::BaseTypeName defaultNestedType)
     {
         refract::MemberElement* element = new refract::MemberElement;
-        refract::IElement* value = RefractElementFromValue<T>(property, defaultNestedType);
+        refract::IElement* value = RefractElementFromValue<T>(SectionInfo<mson::ValueMember>(property.section, property.sourceMap), defaultNestedType);
+
+        if (!property.section.name.literal.empty()) {
+            snowcrash::SourceMap<mson::Literal> sourceMap;
+            sourceMap.sourceMap.append(property.sourceMap.name.sourceMap);
+
+            refract::IElement* key = PrimitiveToRefract(MakeSectionInfo(property.section.name.literal, sourceMap, property.hasSourceMap()));
 
-        if (!property.name.literal.empty()) {
-            element->set(property.name.literal, value);
+            element->set(key, value);
         }
-        else if (!property.name.variable.values.empty()) {
+        else if (!property.section.name.variable.values.empty()) {
 
-            if (property.name.variable.values.size() > 1) {
+            if (property.section.name.variable.values.size() > 1) {
                 // FIXME: is there example for multiple variables?
                 throw snowcrash::Error("multiple variables in property definition are not allowed", snowcrash::MSONError);
             }
 
-            element->set(property.name.variable.values.begin()->literal, value);
-            element->value.first->attributes[SerializeKey::Variable] = refract::IElement::Create(true);
-            SetElementType(element->value.first, property.name.variable.typeDefinition);
+            snowcrash::SourceMap<mson::Literal> sourceMap;
+            sourceMap.sourceMap.append(property.sourceMap.name.sourceMap);
+
+            refract::IElement* key = PrimitiveToRefract(MakeSectionInfo(property.section.name.variable.values.begin()->literal, sourceMap, property.hasSourceMap()));
+
+            key->attributes[SerializeKey::Variable] = refract::IElement::Create(true);
+
+            element->set(key, value);
+
+            SetElementType(element->value.first, property.section.name.variable.typeDefinition);
         }
         else {
             throw snowcrash::Error("no property name", snowcrash::MSONError);
         }
 
-        mson::TypeAttributes attrs = property.valueDefinition.typeDefinition.attributes;
+        mson::TypeAttributes attrs = property.section.valueDefinition.typeDefinition.attributes;
+        // there is no source map for attributes
```

Comment:
What I found i wrote in comment. There is sourceMap for full `mson::ValueDefinition` for `typeSection` part i did not found.
If you know where is sourceMap for typeAttributes link it please.